src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -429,13 +429,34 @@ private async Task<Expression> RecurseChainedExpression(ChainedExpression expres
 
             try
             {
-                (IReadOnlyList<T> results, string nextContinuationToken) = await _fhirDataStore.ExecuteDocumentQueryAsync<T>(sqlQuerySpec, feedOptions, continuationToken, searchOptions.MaxItemCountSpecifiedByClient, feedOptions.MaxConcurrency == null ? searchEnumerationTimeoutOverrideIfSequential : null, cancellationToken);
+                (IReadOnlyList<T> results, string nextContinuationToken) = (null, null);
+                var desiredItemCount = feedOptions.MaxItemCount * CosmosFhirDataStore.ExecuteDocumentQueryAsyncMinimumFillFactor;
 
-                if (queryPartitionStatistics != null && messagesList != null)
+                // Set timeout for sequential query execution
+                if (feedOptions.MaxConcurrency == null &&
+                    _cosmosConfig.ParallelQueryOptions.EnableConcurrencyIfQueryExceedsTimeLimit == true &&
+                    string.IsNullOrEmpty(searchOptions.ContinuationToken))
```

Comment:
Do we need to check searchEnumerationTimeoutOverrideIfSequential also?
```suggestion
                if (feedOptions.MaxConcurrency == null &&
                    _cosmosConfig.ParallelQueryOptions.EnableConcurrencyIfQueryExceedsTimeLimit == true &&
                    string.IsNullOrEmpty(searchOptions.ContinuationToken) &&
                    searchEnumerationTimeoutOverrideIfSequential == null)
```